go/mysql/sql_error.go

```diff
@@ -161,3 +161,12 @@ func NewSQLErrorFromError(err error) error {
 	}
 	return serr
 }
+
+var isGRPCOverflowRE = regexp.MustCompile(`.*grpc: received message larger than max \(\d+ vs. \d+\)`)
+
+func demuxResourceExhaustedErrors(msg string) int {
```

Comment:
TODO: In addition to processing returned errors we should probably also follow this pattern when we construct payload size errors - https://github.com/vitessio/vitess/blob/master/go/vt/vtgate/executor.go#L1131

Reply:
Ah, actually I believe because these (seem to be) expecting a `vtrpc.Code` style error instead of SQL code we can't actually do this here.

Reply:
I'm just having a conversation with myself at this point bc I like the github format for exploring the PR.

But I believe we should expand the `NewSQLErrorFromError` to handle these cases instead. Since (presumably) everything that gets adapted comes back through there it should be a valid chokepoint for converting these.